packages/blade/package.json

```diff
@@ -29,32 +29,32 @@
   "exports": {
     "./components": {
       "react-native": {
-        "import": "./build/components/index.native.js",
-        "types": "./build/components/index.native.d.ts"
+        "types": "./build/components/index.native.d.ts",
+        "default": "./build/components/index.native.js"
```

Comment:
can you link what is the diff between using `default` and `import` here?

Reply:
Whenever Blade is used with `import` (e.g. `import { Button } from "@razorpay/blade/components";`), the tool processing this import (bundler/IDE/test) will look for the "import" condition in Blade's `exports`. This is true for all tools that support properly resolving the "exports" field.

Jest only supports "browser", "require" or "default" as mentioned in https://github.com/okta/okta-auth-js/issues/1268#issue-1331894269, I couldn't find a docs link explaining this. But Jest's support for the "exports" field is still an open issue: https://github.com/browserify/resolve/issues/222

The difference is that `import` will only work if the tool processing it supports it, jest doesn't. The default field instead works as "if nothing else resolves before this, finally resolve to this". If a "default" is not provided, the resolution of the package will fail if the tool doesn't support any of the other defined keys.

Reply:
found it [here](https://nodejs.org/api/packages.html#subpath-exports). Now i understand why jest fails to import because jest still doesn't support importing esm things so we could have done `require` and `import` so jest could use require `export` path and webpack can do `import` export path. but its fine for now since we don't have cjs and esm builds